internal/services/deployer/service.go

```diff
@@ -51,31 +53,46 @@ func (s *Service) Run(ctx context.Context) (err error) {
 	}
 	deployedEntries, err := s.getExternalSystemPoolEntityCount(*systemType)
 	if err != nil {
-		s.log.WithError(err).Warn("unable to get deployed entries count")
+		return errors.Wrap(err, "unable to get deployed entries count")
 	}
+
 	for i := int(deployedEntries); i < s.config.DeployerConfig().ContractCount; i++ {
-		contract, err := s.deployContract()
+		nonce, err := s.eth.PendingNonceAt(ctx, s.config.DeployerConfig().KeyPair.Address())
+		if err != nil {
+			return errors.Wrap(err, "failed to retrieve account nonce")
+		}
+		contractAddress := crypto.CreateAddress(s.config.DeployerConfig().KeyPair.Address(), nonce)
+
+		poolEntryID, err := s.createPoolEntities(contractAddress.Hex(), *systemType)
 		if err != nil {
+			return errors.Wrap(err, "failed to create pool entry")
+		}
+
+		contract, err := s.deployContract(big.NewInt(0).SetUint64(nonce))
+		if err != nil {
+			running.WithThreshold(context.Background(), s.log, "remove-pool", func(ctx context.Context) (bool, error) {
```

Comment:
Why you are not logging WithTreshhold error?

Reply:
WithTreshhold returns nothing, also WithTreshhold logs inside itself
docs : ```// You are generally not supposed to log error inside the runner,
// you should return error instead - non-nil errors returned from runner function
// will be logged with stack and the runnerName.```